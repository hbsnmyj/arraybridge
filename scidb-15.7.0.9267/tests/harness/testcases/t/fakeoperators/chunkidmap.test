--setup
--test
--start-query-logging


load_library('misc')

# test simple 1-d, 2-d, and 3-d cases
create array cim1 <a:int32> [x=0:10000,100,0]
--igdata "store(build(cim1,random()%1000),cim1)"
test_chunk_id_map(cim1)
create array cim2 <a:int32> [x=0:1000,100,0,y=0:5000,100,0]
--igdata "store(build(cim2,random()%1000),cim2)"
test_chunk_id_map(cim2)
create array cim3 <a:int32> [x=0:100,10,0,y=0:500,100,0,z=0:222,50,0]
--igdata "store(build(cim3,random()%1000),cim3)"
test_chunk_id_map(cim3)

# test cases with starting value > 1 and overlap
create array cim4 <a:int32> [x=14:460,30,28]
--igdata "store(build(cim4,random()%1000),cim4)"
test_chunk_id_map(cim4)

# test cases on the border between direct and indirect
create array cim7flat <a:int32, b:int32> [x=0:10000,1000,0]
create array cim7a <a:int32> [x=0:10000,1000,0]
create array cim7b <b:int32> [x=0:10000,1000,0]
--igdata "store(build(cim7a,random()%1000),cim7a)"
--igdata "store(build(cim7b,random()%1000),cim7b)"
--igdata "store(join(cim7a,cim7b), cim7flat)"
create array cim7 <x:int64> [a=0:429496729699,100,0,b=0:429496729499,100,0]
--igdata "redimension(cim7flat, cim7, false)"
test_chunk_id_map(cim7)

# test indirect mapping case
create array cim3n <a:int32> [x=0:100,10,0,y=0:*,100,0,z=0:222,50,0]
--igdata "store(cim3, cim3n)"
test_chunk_id_map(cim3n)

--cleanup

remove(cim1)
remove(cim2)
remove(cim3)
remove(cim4)
remove(cim3n)
remove(cim7)
remove(cim7flat)
remove(cim7a)
remove(cim7b)

--stop-query-logging
