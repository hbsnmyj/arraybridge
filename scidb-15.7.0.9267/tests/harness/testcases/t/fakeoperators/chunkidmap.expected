SCIDB QUERY : <load_library('misc')>
Query was executed successfully

SCIDB QUERY : <create array cim1 <a:int32> [x=0:10000,100,0]>
Query was executed successfully

SCIDB QUERY : <store(build(cim1,random()%1000),cim1)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <test_chunk_id_map(cim1)>
{dummy_dimension} dummy_attribute

SCIDB QUERY : <create array cim2 <a:int32> [x=0:1000,100,0,y=0:5000,100,0]>
Query was executed successfully

SCIDB QUERY : <store(build(cim2,random()%1000),cim2)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <test_chunk_id_map(cim2)>
{dummy_dimension} dummy_attribute

SCIDB QUERY : <create array cim3 <a:int32> [x=0:100,10,0,y=0:500,100,0,z=0:222,50,0]>
Query was executed successfully

SCIDB QUERY : <store(build(cim3,random()%1000),cim3)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <test_chunk_id_map(cim3)>
{dummy_dimension} dummy_attribute

SCIDB QUERY : <create array cim4 <a:int32> [x=14:460,30,28]>
Query was executed successfully

SCIDB QUERY : <store(build(cim4,random()%1000),cim4)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <test_chunk_id_map(cim4)>
{dummy_dimension} dummy_attribute

SCIDB QUERY : <create array cim7flat <a:int32, b:int32> [x=0:10000,1000,0]>
Query was executed successfully

SCIDB QUERY : <create array cim7a <a:int32> [x=0:10000,1000,0]>
Query was executed successfully

SCIDB QUERY : <create array cim7b <b:int32> [x=0:10000,1000,0]>
Query was executed successfully

SCIDB QUERY : <store(build(cim7a,random()%1000),cim7a)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(build(cim7b,random()%1000),cim7b)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <store(join(cim7a,cim7b), cim7flat)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <create array cim7 <x:int64> [a=0:429496729699,100,0,b=0:429496729499,100,0]>
Query was executed successfully

SCIDB QUERY : <redimension(cim7flat, cim7, false)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <test_chunk_id_map(cim7)>
{dummy_dimension} dummy_attribute

SCIDB QUERY : <create array cim3n <a:int32> [x=0:100,10,0,y=0:*,100,0,z=0:222,50,0]>
Query was executed successfully

SCIDB QUERY : <store(cim3, cim3n)>
[Query was executed successfully, ignoring data output by this query.]

SCIDB QUERY : <test_chunk_id_map(cim3n)>
{dummy_dimension} dummy_attribute

SCIDB QUERY : <remove(cim1)>
Query was executed successfully

SCIDB QUERY : <remove(cim2)>
Query was executed successfully

SCIDB QUERY : <remove(cim3)>
Query was executed successfully

SCIDB QUERY : <remove(cim4)>
Query was executed successfully

SCIDB QUERY : <remove(cim3n)>
Query was executed successfully

SCIDB QUERY : <remove(cim7)>
Query was executed successfully

SCIDB QUERY : <remove(cim7flat)>
Query was executed successfully

SCIDB QUERY : <remove(cim7a)>
Query was executed successfully

SCIDB QUERY : <remove(cim7b)>
Query was executed successfully

